pkg/autodiscovery/providers/kubecontainer.go

```diff
@@ -276,46 +213,40 @@ func (k *KubeContainerConfigProvider) generateConfigs() ([]integration.Config, e
 		}
 
 		errs = append(errs, utils.ValidateAnnotationsMatching(
-			pod.Annotations,
+			entity.Annotations,
 			containerIdentifiers,
 			containerNames)...)
 
-		namespacedName := pod.Namespace + "/" + pod.Name
-		for _, err := range errs {
-			if _, found := adErrors[namespacedName]; !found {
-				adErrors[namespacedName] = map[string]struct{}{err.Error(): {}}
-			} else {
-				adErrors[namespacedName][err.Error()] = struct{}{}
-			}
-		}
+	default:
+		log.Errorf("cannot handle entity of kind %s", e.GetID().Kind)
 	}
 
-	bldr := strings.Builder{}
-	for _, c := range configs {
-		fmt.Fprintf(&bldr, "  %s %s %s\n", c.Name, c.Source, c.Digest())
+	if len(errs) > 0 {
+		errMsgSet = make(ErrorMsgSet)
+		for _, err := range errs {
+			errMsgSet[err.Error()] = struct{}{}
+		}
 	}
-	log.Debugf("KubeContainerConfigProvider#generateConfigs generated:\n%s", bldr.String())
-
-	k.configErrors = adErrors
-	telemetry.Errors.Set(float64(len(adErrors)), names.KubeContainer)
 
-	return configs, nil
+	return configs, errMsgSet
 }
 
 // GetConfigErrors returns a map of configuration errors for each namespace/pod
 func (k *KubeContainerConfigProvider) GetConfigErrors() map[string]ErrorMsgSet {
-	k.RLock()
-	defer k.RUnlock()
+	k.mu.RLock()
+	defer k.mu.RUnlock()
```

Comment:
Should this return a copy instead? The caller does not hold the mutex while reading the map, so the map can be concurrently modified by `processEvents`.

Reply:
Indeed, well spotted.